### PR TITLE
standards/inventory-engageya

### PR DESCRIPTION
**Pull request by ricardoribasmrf**
Added engageya for the schemas

### CODE REVIEW

File: json-lint/test/resources/schema/inventory/valid/engageya.test.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "placements": {
+        "nativeAd": {
+            "adServer": "engageya"
+        }
+    },
+    "adServers": {
+        "engageya": {
+            "data-widgetIds": "108404",
```
Comment (joaquimsantmrf) [inline json-lint/test/resources/schema/inventory/valid/engageya.test.json]: int

File: json-lint/test/resources/schema/inventory/valid/engageya.test.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+            "data-websiteId": "154423",
```
Comment (joaquimsantmrf) [inline json-lint/test/resources/schema/inventory/valid/engageya.test.json]: int

File: json-lint/test/resources/schema/inventory/valid/engageya.test.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+            "data-publisherId": "178403",
```
Comment (joaquimsantmrf) [inline json-lint/test/resources/schema/inventory/valid/engageya.test.json]: int

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        "data-publisherId"
+    ],
+    "defaultSnippets": [
+        {
+            "label": "Engageya template",
+            "description": "contains required keys",
+            "body": {
+                "type": "engageya",
+                "layout": "responsive",
+                "data-widgetIds": "$1",
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: int (remove the "")

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                "data-websiteId": "$2",
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: int (remove the "")

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                "data-publisherId": "$3"
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: int (remove the "")

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    "properties": {
+        "type": {
+            "enum": [
+                "engageya"
+            ]
+        },
+        "layout": {
+            "type": "string"
+        },
+        "data-widgetIds": {
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: int or parameter ( look at other adservers)
 type = ["intiger", "string"]
pattern = "\\$\{.*\}"     ((something similar))

Ad tests for both cases!  (make one ad server in the engayaa.test.json with params, one with natural values)

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            "type": "string"
+        },
+        "data-websiteId": {
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: same

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            "type": "string"
+        },
+        "data-publisherId": {
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: same (int/string)

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+                "type": "engageya",
+                "layout": "responsive",
+                "data-widgetIds": "$1",
+                "data-websiteId": "$2",
+                "data-publisherId": "$3"
+            }
+        }
+    ],
+    "properties": {
+        "type": {
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: "type": {"const": "engageya"}  /// check the other adservers, they have all the type like a const

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+            "type": "string"
+        },
+        "width": {
+            "type": [
+                "integer",
+                "string"
+            ],
+            "title": "String of digits ( if not set, default is 300 )",
+            "description": "example: 300",
+            "pattern": "^(\\$\\{.+\\})$"
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: 👍  this patter, also in the other ints up :)

File: json-lint/schemas/inventory/adservers/engageya.json [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    ],
+    "properties": {
+        "type": {
+            "enum": [
+                "engageya"
+            ]
+        },
+        "layout": {
+            "type": "string"
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/engageya.json]: maybe you could have an enum here enum ["responsive", ...]

File: json-lint/test/resources/schema/inventory/valid/engageya.test.json (hunk 1 above)
Comment (ricardoribasmrf) [reply json-lint/test/resources/schema/inventory/valid/engageya.test.json]: In the documentation could be a list of ids separated by comma (id1,id2)

File: json-lint/schemas/inventory/adservers/engageya.json (hunk 12 above)
Comment (ricardoribasmrf) [reply json-lint/schemas/inventory/adservers/engageya.json]: I will add a enum with all the possible types of layout (i will see documentation)

**Comment by joaquimsantmrf** [comment]
"type": {"const": "engageya"}  AND layout as enum
and then 🚀

File: json-lint/schemas/inventory/adservers/types/layout.json [hunk 13]
```diff
@@ -0,0 +1,16 @@
+{
+    "$id": "#layout",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "string",
+    "title": "Layout attribute",
+    "description": "https://www.ampproject.org/docs/design/responsive/control_layout",
+    "additionalProperties": false,
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/types/layout.json]: "type": "string" >> no need of additionalProperties

File: json-lint/schemas/inventory/adservers/types/dimension.json [hunk 14]
```diff
@@ -0,0 +1,12 @@
+{
+    "$id": "#dimension",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": [
+        "integer",
+        "string"
+    ],
+    "title": "Width or Height",
+    "description": "Example: 300",
+    "additionalProperties": false,
```
Comment (joaquimsantmrf) [inline json-lint/schemas/inventory/adservers/types/dimension.json]: "type": "string" >> no need of additionalProperties

**Comment by joaquimsantmrf** [comment]
🚀

**Comment by joaquimsantmrf** [comment]
❤️ last commit

**Comment by ricardoribasmrf** [comment]
Retest this please

**Pull request merged**